### PR TITLE
support a new command line option `systemfile`

### PR DESCRIPTION
**Pull request by ThomasBreuer**
... which allows one to read a given file during GAP's initialization,
after the file `lib/system.g` has been read.

The idea is that at this point, the command line options have been
evaluated, and it is a suitable place for customizing the GAP session
beyond command line options.
The motivation for this proposal is oscar-system/GAP.jl/issues/518,
see the discussion there for the background.

P.S.:
Each of the files `lib/system.g` and `src/system.c` claims that the
command line options listed there must be kept in sync with those listed
in the other file.
Apparently they are aleady out of sync.
If the claims are still relevant then of course
the new command line option should be mentioned also in `src/system.c`.

### CODE REVIEW

**Comment by ChrisJefferson** [comment]
If this is useful to Julia, I'm generally in favour of it.

My only thought is I would either delete the "IsReadableFile" check, or at least put in an error message if the file isn't readable, else at the moment GAP will silently continue if you mispell the filename (which is probably not what we want).

File: lib/system.g [hunk 1]
```diff
@@ -571,3 +573,8 @@ end);
 ##  </ManSection>
 ##
 #T really ???
+
+if GAPInfo.CommandLineOptions.systemfile <> "" and
+   IsReadableFile( GAPInfo.CommandLineOptions.systemfile ) then
+  READ( GAPInfo.CommandLineOptions.systemfile );
```
Comment (fingolfin) [inline lib/system.g:579]:
```suggestion
if GAPInfo.CommandLineOptions.systemfile <> "" and not READ( GAPInfo.CommandLineOptions.systemfile ) then
    Error( "file \"", name, "\" must exist and be readable" );
```

In view of what @ChrisJefferson suggested (and I agree), how about doing something like the above?

**Review by fingolfin — COMMENTED** [review]
Thanks! I left a suggestion for a minor improvement.

Regarding the "keep in sync", I think one has to read these two kernel comments together:
```
/* These are just the options that need kernel processing. Additional options will be 
   recognised and handled in the library */

/* These options must be kept in sync with those in system.g, so the help output
   is correct */
```

So I think that the list in the kernel simply ought to be a subset of the list in the library (unless some command line option is intended to be kept "secret").

**Review by ChrisJefferson — APPROVED** [review]

File: lib/system.g [hunk 2]
```diff
@@ -571,3 +573,10 @@ end);
 ##  </ManSection>
 ##
 #T really ???
+
+if GAPInfo.CommandLineOptions.systemfile <> ""
+   and not READ( GAPInfo.CommandLineOptions.systemfile ) then
+  Print( "Could not read file \"", GAPInfo.CommandLineOptions.systemfile,
+         "\" (command line option --systemfile)\n" );
+  QuitGap();
```
Comment (fingolfin) [inline lib/system.g:581]:
```suggestion
  QuitGap(1);
```
... so that we can detect from scripts that there was an error.

File: lib/system.g [hunk 3]
```diff
@@ -571,3 +573,10 @@ end);
 ##  </ManSection>
 ##
 #T really ???
+
+if GAPInfo.CommandLineOptions.systemfile <> ""
+   and not READ( GAPInfo.CommandLineOptions.systemfile ) then
+  Print( "Could not read file \"", GAPInfo.CommandLineOptions.systemfile,
```
Comment (fingolfin) [inline lib/system.g:579]:
```suggestion
  PRINT_TO( "*errout*", "Could not read file \"", GAPInfo.CommandLineOptions.systemfile,
```

**Review by fingolfin — CHANGES_REQUESTED** [review]
OK, `Error` doesn't work here, but let's still print to errout and set an exit code.

**Review by fingolfin — APPROVED** [review]

**Comment by fingolfin** [comment]
Backported to stable-4.11 in cabff6af41cf0ed4c5b460f5b274d621bb3a3476

**Pull request merged**